aws-inventory/lambda/inventory-elasticbeanstalk.py

```diff
@@ -0,0 +1,122 @@
+        if e.response['Error']['Code'] == 'UnauthorizedOperation':
+            logger.error("Antiope doesn't have proper permissions to this account")
+            return(event)
+        logger.critical("AWS Error getting info for {}: {}".format(message['account_id'], e))
+        capture_error(message, context, e, "ClientError for {}: {}".format(message['account_id'], e))
+        raise
+    except Exception as e:
+        logger.critical("{}\nMessage: {}\nContext: {}".format(e, message, vars(context)))
+        capture_error(message, context, e, "General Exception for {}: {}".format(message['account_id'], e))
+        raise
+
+
+def discover_elastic_beanstalk_applications(target_account:str, region:str):
+    eb = target_account.get_client('elasticbeanstalk', region=region)
+
+    try:
+        apps = eb.describe_applications() # Returns all items no pagination
+    except ClientError as error:
+        logger.error(f"Elastic Beanstalk Describe Application Error: {error}")
+        return
+    for app in apps:
+        resource_item = {
+            "awsAccountId":                 target_account.account_id,
+            "awsAccountName":               target_account.account_name,
+            "resourceType":                 TYPE,
+            "configurationItemCaptureTime": str(datetime.datetime.now()),
+            "awsRegion":                    region,
+            "configuration":                app,
+            "supplementaryConfiguration":   {},
+            "resourceName":                 app.get("ApplicationName"),
+            "resourceId":                   f"{target_account.acount_id}-{region}-{app.get('ApplicationName')}",
+            "ARN":                          app.get("ApplicationArn"),
+            "resourceCreationTime":         app.get("DateCreated"),
+            "errors":                       {}
+        }
+
+        save_resource_to_s3(RESOURCE_PATH, resource_item.get("resourceId"), resource_item)
+
+
+def discover_elastic_beanstalk_environments(target_account:str, region:str):
+    eb = target_account.get_client('elasticbeanstalk', region=region)
+
+    try:
+        envs = eb.describe_environments() # Returns all items no pagination
+    except ClientError as error:
+        logger.error(f"Elastic Beanstalk Describe Environments Error: {error}")
+        return 
+    for env in envs:
+        resource_item = {
+            "awsAccountId":                 target_account.account_id,
+            "awsAccountName":               target_account.account_name,
+            "resourceType":                 TYPE,
+            "configurationItemCaptureTime": str(datetime.datetime.now()),
+            "awsRegion":                    region,
+            "configuration":                env,
+            "supplementaryConfiguration":   {},
+            "resourceName":                 env.get("EnvironmentName"),
+            "resourceId":                   f"{target_account.acount_id}-{region}-{env.get('EnvironmentName')}",
```

Comment:
would `env['EnvironmentId']` work as a unique `resourceId`? If so we should use that